tests/puppeteer/pages/FO/cart.js

```diff
@@ -48,5 +48,20 @@ module.exports = class Cart extends FOBasePage {
    */
   async editProductQuantity(productID, quantity) {
     await this.setValue(this.productQuantity.replace('%NUMBER', productID), quantity);
+    // click on price to see that its changed
+    await this.page.click(this.productPrice.replace('%NUMBER', productID));
+  }
+
+  /**
+   * To get a number from text
+   * @param selector
+   * @param timeout
+   * @return integer
+   */
+  async getPriceFromText(selector, timeout = 0) {
+    await this.page.waitFor(timeout);
+    const text = await this.getTextContent(selector);
+    const number = /[+-]?\d+(\.\d+)?/g.exec(text).toString();
```

Comment:
Hum, depending on how the price if formatted :thinking: 
10 000.99 will be works

Reply:
deleting spaces on string before extracting the number is better maybe ?

Reply:
Depends on what text are returning, give me the string and I'll give you the best regex or code to sanitize it!

Reply:
For now, `text` return something like `€57.41` (no spaces)
With actual regex, it's sure that we will have the value needed `57.41`.
but of course, if text return `€10 000.99`, number will be incorrect `0.99` 
That's why i suggested to remove spaces from text before getting our integer.

Reply:
You could simply use
```suggestion
    const number = Number(text.replace(/[^0-9.-]+/g,''));
```
and everything will be ok :)

Reply:
Shouldn't you escape the dot in your regexp ?

Reply:
nope because it's in `[]`